docs/developers/rollmint.md

```diff
@@ -27,10 +27,375 @@ re-execute the transactions to check the validity of the chain
 (i.e. a full node). Furthermore, Rollmint currently only supports
 a single sequencer.
 
+## Specifications of the Rollmint Scheme and Necessary Components
+
+### Introduction
+
+This document provides an overview on how to implement Cosmos-SDK based Rollup
+chains. It goes into detail about how these chains can be built using Celestia
+as a Data Availability (DA) and Consensus Layer. At the same time we try to
+keep the high level API general enough such that developers can choose
+alternative DA and Consensus Layers (e.g. Ethereum, the Cosmos Hub).
+
+We briefly compare building a Cosmos-SDK Rollmint chain with the status quo
+(building a sovereign Cosmos zone): The current way to use the Cosmos-SDK is to
+define the state-machine, also referred to as the Application. Application
+developers do not have to bother about the lower Layers, namely consensus and
+<!-- is networking right? -->
+networking. The lower Layers are currently handled by another
+application-agnostic piece of the stack, namely the ABCI Client which defaults
+to `tendermint-core`. We will replace the default ABCI Client with one that
+does not handle Consensus itself. Instead it will only provide the networking
+Layers, and will use the Consensus of the DA Layer. How exactly will be
+described below:
+
+To summarize, if a current SDK-based app roughly follows the following
+architecture:
+
+![diagram1.png](/img/rollmint/diagram1.png)
+
+When implementing a Cosmos-SDK based Rollmint chain, this overview changes to:
+
+![diagram2.png](/img/rollmint/diagram2.png)
+
+Note that from the point of view of an application developer, nothing changes.
+Rollmint has the exact same developer experience as the usual Cosmos-SDK with
+Tendermint provides: they still just have to define their business /
+state-machine logic. The Rollmint-specific ABCI Client is stripped down to only
+handle the network Layer but it also embeds a light node of the DA and
+Consensus Layer chain used and takes care of submitting the block to the DA
+and Consensus Layer.
+
+As with Tendermint the state-machine and Rollmint interact with each other via
+ABCI.
+
+#### Definitions
+
+- **Rollmint Block Time** - configurable duration after which a new block of
+the (Rollmint chain) is produced.
+- **Grace Period** - configurable duration between block creation and block
+finality. It’s the time window for fraud proof submission.
+
+#### Architecture Overview
+
+- Rollmint aggregator sends block to main chain for DA and Consensus (Celestia,
+Ethereum, Cosmos Hub)
+- If time passes without fraud proof, the block is considered valid
+- Else, if fraud proof, orphan all subsequent blocks
+- With bond on Ethereum, or with longer block times and p2p gossip in
+side-chains subnet, or, liquidity providers
+
+![lazyledger-diagram.png](/img/rollmint/lazyledger-diagram.png)
```

Comment:
This diagram still implies Optimistic Rollups.

Reply:
yes, noting another for Alina to reproduce